tests/backup/common.go

```diff
@@ -0,0 +1,273 @@
+package tests
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"time"
+
+	api "github.com/portworx/px-backup-api/pkg/apis/v1"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/torpedo/drivers/backup"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/pkg/aetosutil"
+	. "github.com/portworx/torpedo/tests"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	post_install_hook_pod = "pxcentral-post-install-hook"
+	quick_maintenance_pod = "quick-maintenance-repo"
+	full_maintenance_pod  = "full-maintenance-repo"
+	taskNamePrefix        = "backupcreaterestore"
+	defaultTimeout        = 6 * time.Minute
+	orgID                 = "default"
+)
+
+var (
+	create_pre_rule  = false
+	create_post_rule = false
+	// User should keep updating the below 3 datas
+	pre_rule_app   = []string{"cassandra", "postgres"}
+	post_rule_app  = []string{"cassandra"}
+	app_parameters = map[string]map[string]string{
+		"cassandra": {"pre_action_list": "nodetool flush -- keyspace1;", "post_action_list": "nodetool verify -- keyspace1;", "background": "false", "run_in_single_pod": "false"},
+		"postgres":  {"pre_action_list": "PGPASSWORD=$POSTGRES_PASSWORD; psql -U '$POSTGRES_USER' -c 'CHECKPOINT';", "background": "false", "run_in_single_pod": "false"},
+	}
+)
+
+var log *logrus.Logger
+var dash *aetosutil.Dashboard
+var post_rule_uid string
+var pre_rule_uid string
+
+func validateBackupCluster() bool {
+	flag := false
+	labelSelectors := map[string]string{"job-name": post_install_hook_pod}
+	ns := backup.GetPxBackupNamespace()
+	pods, err := core.Instance().GetPods(ns, labelSelectors)
+	if err != nil {
+		log.Errorf("Unable to fetch pxcentral-post-install-hook pod from backup namespace\n Error : [%v]\n",
+			err)
+		return false
+	}
+	for _, pod := range pods.Items {
+		log.Infof("Checking if the pxcentral-post-install-hook pod is in Completed state or not")
+		bkp_pod, err := core.Instance().GetPodByName(pod.GetName(), ns)
+		if err != nil {
+			log.Errorf("An Error Occured while getting the pxcentral-post-install-hook pod details")
```

Comment:
print the error here,
or return the error and handle it in calling function

Reply:
Done!!